Tests/SecureXPCTests/XPCDecoder/Scalar decoding tests.swift

```diff
@@ -0,0 +1,185 @@
+		try assert(xpc_int64_create(               0), decodesEqualTo:    0 as Int64)
+		try assert(xpc_int64_create(             123), decodesEqualTo:  123 as Int64)
+		try assert(xpc_int64_create(Int64(Int64.max)), decodesEqualTo: .max as Int64)
+	}
+	
+	// MARK: Unsigned Integers
+	
+	func testDecodes_XPCUInt64_as_UInt() throws {
+		try assert(xpc_uint64_create(UInt64(UInt.min)), decodesEqualTo: .min as UInt)
+		try assert(xpc_uint64_create(               0), decodesEqualTo:    0 as UInt)
+		try assert(xpc_uint64_create(             123), decodesEqualTo:  123 as UInt)
+		try assert(xpc_uint64_create(UInt64(UInt.max)), decodesEqualTo: .max as UInt)
+	}
+	
+	func testDecodes_XPCUInt64_as_UInt8() throws {
+		try assert(xpc_uint64_create(UInt64(UInt8.min)), decodesEqualTo: .min as UInt8)
+		try assert(xpc_uint64_create(                0), decodesEqualTo:    0 as UInt8)
+		try assert(xpc_uint64_create(              123), decodesEqualTo:  123 as UInt8)
+		try assert(xpc_uint64_create(UInt64(UInt8.max)), decodesEqualTo: .max as UInt8)
+	}
+	
+	func testDecodes_XPCUInt64_as_UInt16() throws {
+		try assert(xpc_uint64_create(UInt64(UInt16.min)), decodesEqualTo: .min as UInt16)
+		try assert(xpc_uint64_create(                 0), decodesEqualTo:    0 as UInt16)
+		try assert(xpc_uint64_create(               123), decodesEqualTo:  123 as UInt16)
+		try assert(xpc_uint64_create(UInt64(UInt16.max)), decodesEqualTo: .max as UInt16)
+	}
+	
+	func testDecodes_XPCUInt64_as_UInt32() throws {
+		try assert(xpc_uint64_create(UInt64(UInt32.min)), decodesEqualTo: .min as UInt32)
+		try assert(xpc_uint64_create(                 0), decodesEqualTo:    0 as UInt32)
+		try assert(xpc_uint64_create(               123), decodesEqualTo:  123 as UInt32)
+		try assert(xpc_uint64_create(UInt64(UInt32.max)), decodesEqualTo: .max as UInt32)
+	}
+	
+	func testDecodes_XPCUInt64_as_UInt64() throws {
+		try assert(xpc_uint64_create(UInt64(UInt64.min)), decodesEqualTo: .min as UInt64)
+		try assert(xpc_uint64_create(                 0), decodesEqualTo:    0 as UInt64)
+		try assert(xpc_uint64_create(               123), decodesEqualTo:  123 as UInt64)
+		try assert(xpc_uint64_create(UInt64(UInt64.max)), decodesEqualTo: .max as UInt64)
+	}
+	
+	// MARK: Floating point numbers
+	
+	func testDecodes_XPCDouble_as_Float() throws {
+		// "Normal" values, from lowest to highest
+		try assert(xpc_double_create(Double(-Float.infinity)),                decodesEqualTo: -Float.infinity)
+		try assert(xpc_double_create(Double(-Float.greatestFiniteMagnitude)), decodesEqualTo: -Float.greatestFiniteMagnitude)
+		try assert(xpc_double_create(Double(-Float.leastNormalMagnitude)),    decodesEqualTo: -Float.leastNormalMagnitude)
+		try assert(xpc_double_create(Double(-Float.leastNonzeroMagnitude)),   decodesEqualTo: -Float.leastNonzeroMagnitude)
+		try assert(xpc_double_create(-123.0),                                 decodesEqualTo: -123.0 as Float)
```

Comment:
Let me try restating what I'm trying to convey, because I think we're actually in agreement here and yet there's a misunderstanding somewhere.

The end-to-end coding for a `Float` starts as a `Float` and ultimately ends as a `Float`. As you expressed this:
`Swift.Float ---encode--> xpc_double_t ---decode--> Swift.Float`

However, for this line of code:
`try assert(xpc_double_create(-123.0),                                 decodesEqualTo: -123.0 as Float)`

What is happening is it starts as a `Double` and ends as a `Float`, or in your way of expressing it:
`Swift.Double ---encode--> xpc_double_t ---decode--> Swift.Float`

I believe this line needs to read:
`try assert(xpc_double_create(Double(-123.0 as Float)),                                 decodesEqualTo: -123.0 as Float)`

Does that make sense?

Reply:
It does, but I that isn't my intention with this test.

The round-trip flow is going to be tested by https://github.com/trilemma-dev/SecureXPC/pull/9. Here I'm trying to achieve this:


```
+----------------------------------------+
| black box that produces a xpc_double_t | --> xpc_double_t ---decode--> Swift.Float
|         with the value 123.0           |
+----------------------------------------+
```

In other words, if I could have written it this way, I would have:

```
let expected: xpc_object_t = 123.0 // With no explicit Double/Float conversion
```

To put another way: I'm treating `xpc_double_create(-123.0)` like a ghetto literal value, in leau of a conformance of `xpc_object_t` to `ExpressibleByFloatLiteral`.

Reply:
Oh I follow now. Yeah this is fine. (This and some other things that have arisen make me want to reconsider how to represent numerics, but that's way outside of the scope of this PR.)